relayer/chain/ethereum/connection.go

```diff
@@ -67,24 +67,26 @@ func (co *Connection) ConnectWithHeartBeat(ctx context.Context, eg *errgroup.Gro
 	co.client = client
 	co.chainID = chainID
 
-	ticker := time.NewTicker(heartBeat)
-
-	eg.Go(func() error {
-		defer ticker.Stop()
-		for {
-			select {
-			case <-ctx.Done():
-				return ctx.Err()
-			case <-ticker.C:
-				_, err := client.NetworkID(ctx)
-				if err != nil {
-					log.WithField("endpoint", co.endpoint).Error("Connection heartbeat failed")
-					return err
+	if heartBeat.Abs() > 0 {
```

Comment:
So if gsrpc has its own ping loop then this one is unnecessary and we can remove it?

Reply:
I assume that the ping loop is specific to WebSocket connections, and there’s no guarantee all service providers will implement it. We can still configure a none-zero heartbeat for HTTP(S) connections or for providers other than Alchemy/Dwellir if necessary.